mono/sgen/sgen-gc.c

```diff
@@ -326,6 +326,15 @@ nursery_canaries_enabled (void)
 
 #define safe_object_get_size	sgen_safe_object_get_size
 
+#if defined(TARGET_OSX) || defined(TARGET_WIN32) || (defined(__linux__) && !defined(TARGET_ANDROID))
```

Comment:
You are right about HOST_WIN32. I was looking for these defines but it seems we are never defining PLATFORM_MACOSX and PLATFORM_LINUX, although this would be what I am looking for.